engines/docker/dockerutils.go

```diff
@@ -0,0 +1,153 @@
+package dockerengine
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/DataDog/zstd"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/pkg/errors"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/fetcher"
+)
+
+type imageFetchContext struct {
+	*runtime.TaskContext
+}
+
+func (ctx imageFetchContext) Progress(description string, percent float64) {
+	ctx.Log(fmt.Sprintf("Progress: %s %f", description, percent))
+}
+
+type dockerClient struct {
+	*docker.Client
+	context imageFetchContext
+}
+
+type tempFileReseter struct {
```

Comment:
You can probably use:
https://godoc.org/github.com/taskcluster/taskcluster-worker/runtime/fetcher#FileReseter

As `TemporaryFile` implements the `fetcher.File` interface.